header/header.go

```diff
@@ -114,9 +114,9 @@ func (eh *ExtendedHeader) Validate() error {
 		return fmt.Errorf("ValidateBasic error on RawHeader at height %d: %w", eh.Height(), err)
 	}
 
-	if eh.RawHeader.Version.App != appconsts.LatestVersion {
-		return fmt.Errorf("app version mismatch, expected: %d, got %d", appconsts.LatestVersion,
-			eh.RawHeader.Version.App)
+	if eh.RawHeader.Version.App > appconsts.LatestVersion {
+		return fmt.Errorf("app version must be less than the latest app version %d, got %d",
+			appconsts.LatestVersion, eh.RawHeader.Version.App)
```

Comment:
```suggestion
		return fmt.Errorf("header received at height %d has version %d, this node supports up to version %d. Please upgrade to support new version",
			eh.RawHeader.Height, eh.RawHeader.Version.App, appconsts.LatestVersion)
```

nit: I would have something a little more descriptive as this will likely get hit by all nodes that don't upgrade to v3 when the consensus network does